app/lib/Plugins/SearchEngine/Elastic8.php

```diff
@@ -306,77 +323,18 @@ public function init() {
 	 * Completely clear index (usually in preparation for a full reindex)
 	 *
 	 * @param null|int $table_num
-	 * @param bool $dont_refresh
 	 *
 	 * @return bool
 	 */
-	public function truncateIndex( $table_num = null, $dont_refresh = false ) {
-		if ( ! $table_num ) {
-			// nuke the entire index
-			try {
-				$mapping = new Elastic8\Mapping();
-
-				foreach ( $mapping->get() as $table => $config ) {
-					$this->getClient()->indices()->delete( [ 'index' => $this->getIndexName( $table ) ] );
-				}
-			} catch ( ClientResponseException $e ) {
-				// noop
-			} //finally {
-			if ( ! $dont_refresh ) {
-				$this->refreshMapping( true );
-			}
-			//}
+	public function truncateIndex( $table_num = null) {
+		$mapping = new Elastic8\Mapping();
+		if ( $table_num ) {
+			$tables = [ Datamodel::getTableName( $table_num ) ];
 		} else {
-			// use scroll API to find all documents in a particular mapping/table and delete them using the bulk API
-			// @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
-			// @see https://www.elastic.co/guide/en/elasticsearch/client/php-api/2.0/_search_operations.html#_scan_scroll
-			$table = Datamodel::getTableName( $table_num );
-			$search_params = [
-				'scroll' => '1m',          // how long between scroll requests. should be small!
-				'index' => $this->getIndexName( $table ),
-				'body' => [
-					'query' => [
-						'match_all' => new stdClass()
-					]
-				]
-			];
-
-			$tmp = $this->getClient()->search( $search_params );   // Execute the search
-			$scroll_id = $tmp['_scroll_id'];   // The response will contain no results, just a _scroll_id
-
-			// Now we loop until the scroll "cursors" are exhausted
-			$delete_params = [];
-			while ( true ) {
-
-				$response = $this->getClient()->scroll( [
-						'scroll_id' => $scroll_id,  //...using our previously obtained _scroll_id
-						'scroll' => '1m'           // and the same timeout window
-					]
-				);
-
-				if ( sizeof( $response['hits']['hits'] ) > 0 ) {
-					foreach ( $response['hits']['hits'] as $result ) {
-						$delete_params['body'][] = [
-							'delete' => [
-								'_index' => $this->getIndexName( $table_num ),
-								'_id' => $result['_id']
-							]
-						];
-					}
-
-					// Must always refresh your _scroll_id!  It can change sometimes
-					$scroll_id = $response['_scroll_id'];
-				} else {
-					// No results, scroll cursor is empty
-					break;
-				}
-			}
-
-			if ( sizeof( $delete_params ) > 0 ) {
-				$this->getClient()->bulk( $delete_params );
-			}
+			$tables = $mapping->getTables();
 		}
-
+		$this->getClient()->indices()->delete( [ 'index' =>  array_map($this->getIndexName, $tables), 'ignore_unavailable' => true ] );
```

Comment:
`$this->getIndexName` should this be something like `$this->getIndexName($table)` or am I missing something else?

Reply:
It's an array_map with the first argument being a callable or a string representing a callable.

Reply:
Changes :
```php
['ca_objects', 'ca_occurrences'...]
// to
['collectiveaccess_ca_objects', 'collectiveaccess_ca_occurrences'...]
```

Reply:
but there isn't a property called getIndexName? there is a function with that name?

Reply:
![image](https://github.com/gaiaresources/providence/assets/53800058/236b7531-2c06-4dd3-8ffb-9c647fb679bc)

Reply:
Can you replace it with:
```php
$this->getClient()->indices()->delete( [ 'index' =>  array_map('getIndexName', $tables), 'ignore_unavailable' => true ] );
```